### PR TITLE
feat(theme): added dark theme

### PR DESCRIPTION
**Pull request by mcoker**
**Note:** This is not production ready and we are not advising products to adopt the dark theme stylesheet at this time. This feature is being released to test in specific use cases to assess the feasibility of adding a dark theme to PatternFly.

--

Adds `patternfly-theme-dark.css`
* This stylesheet is intended to be loaded after all PatternFly styles (base, components, layouts, etc), but before any existing overrides.
* Dark theme styles are enabled by adding the class `pf-theme-dark` to the `<html>` tag.
* The dark theme overrides use the selector `:where(.pf-theme-dark)` so that the use of the additional class to target dark theme overrides does not increase the specificity of the overrides.
  * Generally speaking, this works better for:
    * Making small adjustments to components for dark theme styles without introducing new specificity issues.
    * Ensuring most existing overrides/customizations on top of PatternFly should still continue to work as long as they're loaded after the dark theme styles from PatternFly.

### CODE REVIEW

**Review by evwilkin — APPROVED** [review]
LGTM - thanks @mcoker! 🎉

**Review by DaoDaoNoCode — COMMENTED** [review]
Screen Shot 2022-02-10 at 11 29 07 AM
Looks like the demos are broken, any ideas?

**Comment by mcoker** [comment]
> Looks like the demos are broken, any ideas?

@DaoDaoNoCode oh good catch. That's a separate issue. Just rebased - should fix it.

**Comment by evwilkin** [comment]
> > Looks like the demos are broken, any ideas?
> 
> @DaoDaoNoCode oh good catch. That's a separate issue. Just rebased - should fix it.

Looks like the rebase fixed this issue 👏  though FYI the preview link is missing a screenshot for the first `Page` demo now (https://patternfly-pr-4654.surge.sh/components/page/html-demos#basic)

**Comment by mcoker** [comment]
@evwilkin awesome! I regenerate all screenshots in the release notes PR, so that should be fine.

File: src/patternfly/patternfly-theme-dark.scss [hunk 1]
```diff
@@ -7,3 +7,9 @@ $pf-theme-dark-class: ".pf-theme-dark" !default;
 :where(#{$pf-theme-dark-class}) {
   @include pf-theme-dark($pf-theme-dark-class);
 }
+
+/* stylelint-disable */
+.ws-preview,
+.ws-example {
```
Comment (DaoDaoNoCode) [inline src/patternfly/patternfly-theme-dark.scss:13]: Maybe also add `.ws-code` here? So the doc will look nicer.

Comment (mcoker) [reply src/patternfly/patternfly-theme-dark.scss:13]: Oh I'm just adding these temporarily and was planning on taking them out. Ideally I think we would put dark theme workspace changes in the patternfly-org repo where the workspace/website theme styles are hosted.

**Pull request merged**